### PR TITLE
mknod: Use ArgsParser and add option for specifying file permissions

### PR DESCRIPTION
**Pull request by monroeclinton**
This patch fixes the todo of implement file permissions for the `mknod` command.

https://man7.org/linux/man-pages/man1/mknod.1.html

### CODE REVIEW

File: Userland/Utilities/mknod.cpp [hunk 1]
```diff
@@ -5,36 +5,36 @@
  */
```
Comment (GMTA) [inline Userland/Utilities/mknod.cpp:6]: Commit message: what is `ParseArgs`? Should probably be `ArgsParser`

File: Userland/Utilities/mknod.cpp [hunk 2]
```diff
@@ -5,36 +5,36 @@
...
 
+    StringView name;
+    StringView type_string;
+    StringView major_string;
+    StringView minor_string;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("Creates a file system node.");
+    args_parser.add_positional_argument(name, "Pathname to create", "name", Core::ArgsParser::Required::Yes);
+    args_parser.add_positional_argument(type_string, "Type of file to create", "type", Core::ArgsParser::Required::Yes);
```
Comment (GMTA) [inline Userland/Utilities/mknod.cpp:31]: The help output lost its ability to communicate which types are supported (`c|b|p`). Maybe add this back to the description?

File: Userland/Utilities/mknod.cpp [hunk 3]
```diff
@@ -1,59 +1,81 @@
...
+            warnln("Do not set device numbers when creating FIFO");
+            return 1;
+        }
+    } else {
+        if (major_string.is_empty() || minor_string.is_empty()) {
+            warnln("Major and minor device numbers are required");
+            return 1;
+        }
+
+        major = atoi(major_string.characters_without_null_termination());
```
Comment (tcl3) [inline Userland/Utilities/mknod.cpp:77]: I think using `to_int()` would be better here, as `atoi()` requires its input to be null-terminated.

File: Userland/Utilities/mknod.cpp [hunk 4]
```diff
@@ -1,59 +1,80 @@
...
+            return 1;
+        }
+    } else {
+        if (!has_value) {
+            warnln("Major and minor device numbers are required");
+            return 1;
+        }
     }
 
     TRY(Core::System::mknod(name, mode, makedev(major, minor)));
```
Comment (tcl3) [inline Userland/Utilities/mknod.cpp:80]: Very minor nitpick: I think you can make the code a bit more concise by rearranging it slightly. Something like:
```suggestion
    auto maybe_major = major_string.to_int();
    auto maybe_minor = minor_string.to_int();
    dev_t device;
    if (type == 'p') {
        if (maybe_major.has_value() || maybe_minor.has_value()) {
            warnln("Do not set device numbers when creating FIFO");
            return 1;
        }
        device = makedev(0, 0);
    } else {
        if (!(maybe_major.has_value() && maybe_minor.has_value())) {
            warnln("Major and minor device numbers are required");
            return 1;
        }
        device = makedev(maybe_major.value(), maybe_minor.value());
    }

    TRY(Core::System::mknod(name, mode, device));
```

Comment (monroeclinton) [reply Userland/Utilities/mknod.cpp]: Good suggestion, thanks

**Review by GMTA — APPROVED** [review]

**Pull request merged**